diff --git a/src/backend/utils/misc/stack_depth.c b/src/backend/utils/misc/stack_depth.c
index 8f7cf531fb..f437ff4d0b 100644
--- a/src/backend/utils/misc/stack_depth.c
+++ b/src/backend/utils/misc/stack_depth.c
@@ -175,7 +175,7 @@ assign_max_stack_depth(int newval, void *extra)
 ssize_t
 get_stack_depth_rlimit(void)
 {
-#if defined(HAVE_GETRLIMIT)
+#if defined(HAVE_GETRLIMIT) && !defined(__wasi__)
 	static ssize_t val = 0;
 
 	/* This won't change after process launch, so check just once */
@@ -194,8 +194,10 @@ get_stack_depth_rlimit(void)
 			val = rlim.rlim_cur;
 	}
 	return val;
-#else
+#elif defined(WIN32)
 	/* On Windows we set the backend stack size in src/backend/Makefile */
 	return WIN32_STACK_RLIMIT;
+#else
+	return -1;
 #endif
 }
