diff --git a/GNUmakefile.in b/GNUmakefile.in
index c1d58738f4..90bb87241d 100644
--- a/GNUmakefile.in
+++ b/GNUmakefile.in
@@ -9,7 +9,7 @@ SUBDIRS = liblwgeom
 ifeq (@SUPPORT_POSTGRESQL@,yes)
 SUBDIRS += @DEPS_SUBDIR@ libpgcommon postgis regress @TOPOLOGY@ @SFCGAL@ doc
 endif
-SUBDIRS += @RASTER@ loader
+SUBDIRS += @RASTER@
 ifeq (@SUPPORT_POSTGRESQL@,yes)
 # NOTE: utils need be built after raster
 # See https://trac.osgeo.org/postgis/ticket/5779
