diff --git a/src/backend/archive/shell_archive.c b/src/backend/archive/shell_archive.c
index 828723afe4..1b2d975463 100644
--- a/src/backend/archive/shell_archive.c
+++ b/src/backend/archive/shell_archive.c
@@ -15,7 +15,9 @@
  */
 #include "postgres.h"
 
+#ifndef __wasi__
 #include <sys/wait.h>
+#endif
 
 #include "access/xlog.h"
 #include "archive/archive_module.h"
@@ -45,18 +47,26 @@ shell_archive_init(void)
 static bool
 shell_archive_configured(ArchiveModuleState *state)
 {
+#ifdef __wasi__
+	arch_module_check_errdetail("WASI does not support shell archive commands.");
+	return false;
+#else
 	if (XLogArchiveCommand[0] != '\0')
 		return true;
 
 	arch_module_check_errdetail("\"%s\" is not set.",
 								"archive_command");
 	return false;
+#endif
 }
 
 static bool
 shell_archive_file(ArchiveModuleState *state, const char *file,
 				   const char *path)
 {
+#ifdef __wasi__
+	return false;
+#else
 	char	   *xlogarchcmd;
 	char	   *nativePath = NULL;
 	int			rc;
@@ -133,6 +143,7 @@ shell_archive_file(ArchiveModuleState *state, const char *file,
 
 	elog(DEBUG1, "archived write-ahead log file \"%s\"", file);
 	return true;
+#endif
 }
 
 static void
