diff --git a/src/backend/postmaster/postmaster.c b/src/backend/postmaster/postmaster.c
index e01d9f0cfe..5689a37f7c 100644
--- a/src/backend/postmaster/postmaster.c
+++ b/src/backend/postmaster/postmaster.c
@@ -74,7 +74,9 @@
 #include <sys/socket.h>
 #include <fcntl.h>
 #include <sys/param.h>
+#ifndef __wasi__
 #include <netdb.h>
+#endif
 #include <limits.h>
 
 #ifdef USE_BONJOUR
@@ -4577,7 +4579,9 @@ pgwin32_register_deadchild_callback(HANDLE procHandle, DWORD procId)
 static void
 InitPostmasterDeathWatchHandle(void)
 {
-#ifndef WIN32
+#if defined(__wasi__)
+	return;
+#elif !defined(WIN32)
 
 	/*
 	 * Create a pipe. Postmaster holds the write end of the pipe open
