diff --git a/src/backend/postmaster/syslogger.c b/src/backend/postmaster/syslogger.c
index 50c2edec1f..6fc776826f 100644
--- a/src/backend/postmaster/syslogger.c
+++ b/src/backend/postmaster/syslogger.c
@@ -52,6 +52,54 @@
 #include "utils/memutils.h"
 #include "utils/ps_status.h"
 
+#ifdef __wasi__
+
+bool		Logging_collector = false;
+int			Log_RotationAge = HOURS_PER_DAY * MINS_PER_HOUR;
+int			Log_RotationSize = 10 * 1024;
+char	   *Log_directory = NULL;
+char	   *Log_filename = NULL;
+bool		Log_truncate_on_rotation = false;
+int			Log_file_mode = S_IRUSR | S_IWUSR;
+NON_EXEC_STATIC pg_time_t first_syslogger_file_time = 0;
+int			syslogPipe[2] = {-1, -1};
+
+int
+SysLogger_Start(int child_slot)
+{
+	(void) child_slot;
+	return 0;
+}
+
+void
+write_syslogger_file(const char *buffer, int count, int destination)
+{
+	(void) buffer;
+	(void) count;
+	(void) destination;
+}
+
+void
+SysLoggerMain(const void *startup_data, size_t startup_data_len)
+{
+	(void) startup_data;
+	(void) startup_data_len;
+	proc_exit(0);
+}
+
+bool
+CheckLogrotateSignal(void)
+{
+	return false;
+}
+
+void
+RemoveLogrotateSignalFiles(void)
+{
+}
+
+#else
+
 /*
  * We read() into a temp buffer twice as big as a chunk, so that any fragment
  * left after processing can be moved down to the front and we'll still have
@@ -1597,3 +1645,5 @@ sigUsr1Handler(SIGNAL_ARGS)
 	rotation_requested = true;
 	SetLatch(MyLatch);
 }
+
+#endif
