diff --git a/src/include/replication/walreceiver.h b/src/include/replication/walreceiver.h
index 89f63f908f..b1c0f0a388 100644
--- a/src/include/replication/walreceiver.h
+++ b/src/include/replication/walreceiver.h
@@ -12,7 +12,13 @@
 #ifndef _WALRECEIVER_H
 #define _WALRECEIVER_H
 
+#ifndef __wasi__
 #include <netdb.h>
+#else
+#ifndef NI_MAXHOST
+#define NI_MAXHOST 1025
+#endif
+#endif
 
 #include "access/xlog.h"
 #include "access/xlogdefs.h"
