diff --git a/src/include/libpq/pqcomm.h b/src/include/libpq/pqcomm.h
index f04ca13565..859116cc35 100644
--- a/src/include/libpq/pqcomm.h
+++ b/src/include/libpq/pqcomm.h
@@ -18,7 +18,9 @@
 
 #include <sys/socket.h>
 #include <sys/un.h>
+#ifndef __wasi__
 #include <netdb.h>
+#endif
 #include <netinet/in.h>
 
 /*
@@ -57,7 +59,11 @@ typedef struct
  * a useful way when the socket pathname is too long, we have to test for
  * this explicitly, instead of just letting the subroutine return an error.)
  */
+#ifdef __wasi__
+#define UNIXSOCK_PATH_BUFLEN MAXPGPATH
+#else
 #define UNIXSOCK_PATH_BUFLEN sizeof(((struct sockaddr_un *) NULL)->sun_path)
+#endif
 
 /*
  * A host that looks either like an absolute path or starts with @ is
