diff --git a/src/include/utils/relcache.h b/src/include/utils/relcache.h
index 3561c6bef0..338feea41e 100644
--- a/src/include/utils/relcache.h
+++ b/src/include/utils/relcache.h
@@ -47,6 +47,9 @@ AssertCouldGetRelation(void)
 #endif
 extern Relation RelationIdGetRelation(Oid relationId);
 extern void RelationClose(Relation relation);
+#ifdef __wasi__
+extern void pglite_reset_relation_cache(void);
+#endif
 
 /*
  * Routines to compute/retrieve additional cached information
