diff --git a/src/include/port.h b/src/include/port.h
index caf65cef5c..8c7f764b9d 100644
--- a/src/include/port.h
+++ b/src/include/port.h
@@ -119,6 +119,7 @@ extern void pgfnames_cleanup(char **filenames);
  * are actually reporting errors typically single out EPIPE and ECONNRESET,
  * while allowing the network failures to be reported generically.
  */
+#ifdef EHOSTDOWN
 #define ALL_CONNECTION_FAILURE_ERRNOS \
 	EPIPE: \
 	case ECONNRESET: \
@@ -129,6 +130,17 @@ extern void pgfnames_cleanup(char **filenames);
 	case ENETRESET: \
 	case ENETUNREACH: \
 	case ETIMEDOUT
+#else
+#define ALL_CONNECTION_FAILURE_ERRNOS \
+	EPIPE: \
+	case ECONNRESET: \
+	case ECONNABORTED: \
+	case EHOSTUNREACH: \
+	case ENETDOWN: \
+	case ENETRESET: \
+	case ENETUNREACH: \
+	case ETIMEDOUT
+#endif
 
 /* Portable locale initialization (in exec.c) */
 extern void set_pglocale_pgservice(const char *argv0, const char *app);
