diff --git a/src/segment/io.h b/src/segment/io.h
index 7835a0c5..ab72156f 100644
--- a/src/segment/io.h
+++ b/src/segment/io.h
@@ -70,20 +70,16 @@ typedef struct TpSegmentWriter
 	uint32			  posting_buffer_size; /* Current size of buffer */
 } TpSegmentWriter;
 
-/* Forward declarations */
-struct TermInfo;
-struct TpDocMapBuilder;
+/* Forward declarations for index.c */
+struct TpLocalIndexState;
 
 /*
  * Function declarations
  */
 
 /* Writer functions */
-extern BlockNumber tp_write_segment(
-		Relation				index,
-		struct TermInfo		   *terms,
-		uint32					num_terms,
-		struct TpDocMapBuilder *docmap);
+extern BlockNumber
+			tp_write_segment(struct TpLocalIndexState *state, Relation index);
 extern void tp_segment_writer_init(TpSegmentWriter *writer, Relation index);
 extern void
 tp_segment_writer_write(TpSegmentWriter *writer, const void *data, uint32 len);
