diff --git a/contrib/pg_freespacemap/pg_freespacemap--1.1.sql b/contrib/pg_freespacemap/pg_freespacemap--1.1.sql
index e1b8242268..b4ddba52e7 100644
--- a/contrib/pg_freespacemap/pg_freespacemap--1.1.sql
+++ b/contrib/pg_freespacemap/pg_freespacemap--1.1.sql
@@ -13,11 +13,8 @@ LANGUAGE C STRICT PARALLEL SAFE;
 CREATE FUNCTION
   pg_freespace(rel regclass, blkno OUT bigint, avail OUT int2)
 RETURNS SETOF RECORD
-AS $$
-  SELECT blkno, pg_freespace($1, blkno) AS avail
-  FROM generate_series(0, pg_relation_size($1) / current_setting('block_size')::bigint - 1) AS blkno;
-$$
-LANGUAGE SQL PARALLEL SAFE;
+AS 'MODULE_PATHNAME', 'pg_freespace_rel'
+LANGUAGE C STRICT PARALLEL SAFE;
 
 
 -- Don't want these to be available to public.
