diff --git a/src/common/exec.c b/src/common/exec.c
index 8b690a1018..5269092765 100644
--- a/src/common/exec.c
+++ b/src/common/exec.c
@@ -31,7 +31,9 @@
 
 #include <signal.h>
 #include <sys/stat.h>
+#ifndef __wasi__
 #include <sys/wait.h>
+#endif
 #include <unistd.h>
 
 #ifdef EXEC_BACKEND
@@ -351,6 +353,12 @@ find_other_exec(const char *argv0, const char *target,
 char *
 pipe_read_line(char *cmd)
 {
+#ifdef __wasi__
+	errno = ENOSYS;
+	log_error(errcode(ERRCODE_SYSTEM_ERROR),
+			  _("could not execute command \"%s\": %m"), cmd);
+	return NULL;
+#else
 	FILE	   *pipe_cmd;
 	char	   *line;
 
@@ -381,6 +389,7 @@ pipe_read_line(char *cmd)
 	(void) pclose_check(pipe_cmd);
 
 	return line;
+#endif
 }
 
 
@@ -390,6 +399,19 @@ pipe_read_line(char *cmd)
 int
 pclose_check(FILE *stream)
 {
+#ifdef __wasi__
+	extern int	pgl_pclose(FILE *stream);
+	int			exitstatus;
+
+	exitstatus = pgl_pclose(stream);
+
+	if (exitstatus == 0)
+		return 0;
+
+	log_error(errcode(ERRCODE_SYSTEM_ERROR),
+			  _("%s() failed: %m"), "pclose");
+	return exitstatus;
+#else
 	int			exitstatus;
 	char	   *reason;
 
@@ -412,6 +434,7 @@ pclose_check(FILE *stream)
 		pfree(reason);
 	}
 	return exitstatus;
+#endif
 }
 
 /*
