diff --git a/src/common/wait_error.c b/src/common/wait_error.c
index 1477887dee..3d75b1d2c4 100644
--- a/src/common/wait_error.c
+++ b/src/common/wait_error.c
@@ -21,7 +21,9 @@
 #endif
 
 #include <signal.h>
+#ifndef __wasi__
 #include <sys/wait.h>
+#endif
 
 /*
  * Return a human-readable string explaining the reason a child process
@@ -34,6 +36,15 @@ wait_result_to_str(int exitstatus)
 {
 	char		str[512];
 
+#ifdef __wasi__
+	if (exitstatus == -1)
+		snprintf(str, sizeof(str), "%m");
+	else
+		snprintf(str, sizeof(str),
+				 _("child process exited with unrecognized status %d"),
+				 exitstatus);
+	return pstrdup(str);
+#else
 	/*
 	 * To simplify using this after pclose() and system(), handle status -1
 	 * first.  In that case, there is no wait result but some error indicated
@@ -83,6 +94,7 @@ wait_result_to_str(int exitstatus)
 				 exitstatus);
 
 	return pstrdup(str);
+#endif
 }
 
 /*
@@ -101,11 +113,17 @@ wait_result_to_str(int exitstatus)
 bool
 wait_result_is_signal(int exit_status, int signum)
 {
+#ifdef __wasi__
+	(void) exit_status;
+	(void) signum;
+	return false;
+#else
 	if (WIFSIGNALED(exit_status) && WTERMSIG(exit_status) == signum)
 		return true;
 	if (WIFEXITED(exit_status) && WEXITSTATUS(exit_status) == 128 + signum)
 		return true;
 	return false;
+#endif
 }
 
 /*
@@ -120,12 +138,18 @@ wait_result_is_signal(int exit_status, int signum)
 bool
 wait_result_is_any_signal(int exit_status, bool include_command_not_found)
 {
+#ifdef __wasi__
+	(void) exit_status;
+	(void) include_command_not_found;
+	return false;
+#else
 	if (WIFSIGNALED(exit_status))
 		return true;
 	if (WIFEXITED(exit_status) &&
 		WEXITSTATUS(exit_status) > (include_command_not_found ? 125 : 128))
 		return true;
 	return false;
+#endif
 }
 
 /*
@@ -137,6 +161,9 @@ wait_result_is_any_signal(int exit_status, bool include_command_not_found)
 int
 wait_result_to_exit_code(int exit_status)
 {
+#ifdef __wasi__
+	return exit_status == -1 ? -1 : 1;
+#else
 	if (exit_status == -1)
 		return -1;				/* failure of pclose() or system() */
 	if (WIFEXITED(exit_status))
@@ -145,4 +172,5 @@ wait_result_to_exit_code(int exit_status)
 		return 128 + WTERMSIG(exit_status);
 	/* On many systems, this is unreachable */
 	return -1;
+#endif
 }
