diff --git a/raster/Makefile.in b/raster/Makefile.in
index 504b80dd6c..da4b67fed8 100644
--- a/raster/Makefile.in
+++ b/raster/Makefile.in
@@ -29,7 +29,7 @@ all:
 ifdef with_pg
 	$(MAKE) @RT_PG_LIB@
 endif
-	$(MAKE) @RT_LOADER@
+	#$(MAKE) @RT_LOADER@
 ifdef with_pg
 	$(MAKE) @RT_POSTGIS_SQL@
 endif
@@ -47,7 +47,7 @@ rtpostgis.sql:
 	$(MAKE) -C rt_pg rtpostgis.sql
 
 install: all
-	$(MAKE) -C loader install
+	#$(MAKE) -C loader install
 ifdef with_pg
 	$(MAKE) -C rt_pg install
 	$(MAKE) -C scripts install
