diff --git a/src/backend/utils/adt/version.c b/src/backend/utils/adt/version.c
index 9f56ef3fb8..48cf7317a5 100644
--- a/src/backend/utils/adt/version.c
+++ b/src/backend/utils/adt/version.c
@@ -16,9 +16,19 @@
 
 #include "utils/builtins.h"
 
+#ifdef __PGLITE__
+#ifndef PGLITE_VERSION
+#define PGLITE_VERSION "0.5.3"
+#endif
+#endif
+
 
 Datum
 pgsql_version(PG_FUNCTION_ARGS)
 {
+#ifdef __PGLITE__
+	PG_RETURN_TEXT_P(cstring_to_text("PostgreSQL " PG_VERSION " (PGlite " PGLITE_VERSION ")"));
+#else
 	PG_RETURN_TEXT_P(cstring_to_text(PG_VERSION_STR));
+#endif
 }
