diff --git a/src/backend/utils/init/miscinit.c b/src/backend/utils/init/miscinit.c
index 8648b8ef42..907649d671 100644
--- a/src/backend/utils/init/miscinit.c
+++ b/src/backend/utils/init/miscinit.c
@@ -202,9 +202,13 @@ InitStandaloneProcess(const char *argv0)
 	/* Compute paths, no postmaster to inherit from */
 	if (my_exec_path[0] == '\0')
 	{
+#ifdef __wasi__
+		strlcpy(my_exec_path, "/pglite/bin/postgres", MAXPGPATH);
+#else
 		if (find_my_exec(argv0, my_exec_path) < 0)
 			elog(FATAL, "%s: could not locate my own executable path",
 				 argv0);
+#endif
 	}
 
 	if (pkglib_path[0] == '\0')
