diff --git a/src/include/storage/fd.h b/src/include/storage/fd.h
index 75aee33b11..645e0eb56a 100644
--- a/src/include/storage/fd.h
+++ b/src/include/storage/fd.h
@@ -178,6 +178,9 @@ extern int	MakePGDirectory(const char *directoryName);
 /* Miscellaneous support routines */
 extern void InitFileAccess(void);
 extern void InitTemporaryFileAccess(void);
+#ifdef __wasi__
+extern void pglite_reset_file_access(void);
+#endif
 extern void set_max_safe_fds(void);
 extern void closeAllVfds(void);
 extern void SetTempTablespaces(Oid *tableSpaces, int numSpaces);
