diff --git a/src/template/wasi b/src/template/wasi
new file mode 100644
index 0000000000..026adfb9a2
--- /dev/null
+++ b/src/template/wasi
@@ -0,0 +1,9 @@
+# src/template/wasi
+
+# Prefer unnamed POSIX semaphores if available, unless user overrides choice.
+if test x"$PREFERRED_SEMAPHORES" = x"" ; then
+  PREFERRED_SEMAPHORES=UNNAMED_POSIX
+fi
+
+CPPFLAGS="$CPPFLAGS -D_GNU_SOURCE -D__PGLITE__"
+CFLAGS_SL="-fPIC"
