diff --git a/src/backend/libpq/ifaddr.c b/src/backend/libpq/ifaddr.c
index 1303a4543e..5ac36dd301 100644
--- a/src/backend/libpq/ifaddr.c
+++ b/src/backend/libpq/ifaddr.c
@@ -22,7 +22,9 @@
 #include <unistd.h>
 #include <sys/stat.h>
 #include <sys/socket.h>
+#ifndef __wasi__
 #include <netdb.h>
+#endif
 #include <netinet/in.h>
 #include <netinet/tcp.h>
 #include <sys/file.h>
@@ -215,7 +217,15 @@ run_ifaddr_callback(PgIfAddrCallback callback, void *cb_data,
 	(*callback) (addr, mask, cb_data);
 }
 
-#ifdef WIN32
+#ifdef __wasi__
+
+int
+pg_foreach_ifaddr(PgIfAddrCallback callback, void *cb_data)
+{
+	return 0;
+}
+
+#elif defined(WIN32)
 
 #include <winsock2.h>
 #include <ws2tcpip.h>
